src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -426,6 +435,37 @@ private void EnqueueDocumentChangedEvent(Solution oldSolution, Solution newSolut
                         var newProject = newSolution.GetRequiredProject(documentId.ProjectId);
 
                         await EnqueueChangedDocumentWorkItemAsync(oldProject.GetRequiredDocument(documentId), newProject.GetRequiredDocument(documentId)).ConfigureAwait(false);
+
+                        if (WorkspaceConfigurationOptions.ShouldConnectSourceGeneratedFilesToWorkspace(newSolution.Options))
+                        {
+                            var oldProjectSourceGeneratedDocuments = await oldProject.GetSourceGeneratedDocumentsAsync(_shutdownToken).ConfigureAwait(false);
+                            var oldProjectSourceGeneratedDocumentsById = oldProjectSourceGeneratedDocuments.ToDictionary(static document => document.Id);
```

Comment:
Ironically we already have a dictionary keyed by ID under the covers here. I'm not sure this is really a hotspot but might be worth sticking a hint to somebody if ever that _does_ come up that there's a way to lessen this.

Reply:
⏱️ Will add comment

Reply:
➡️ Now updated